app/controllers/shopping_carts_controller.rb

```diff
@@ -2,8 +2,20 @@ class ShoppingCartsController < ApplicationController
   before_action :get_shopping_cart
 
   def complete
```

Comment:
This method already looks too long and too nested. Can we have another method for rating to be called within complete action?